include/ensmallen_bits/callbacks/progress_bar.hpp

```diff
@@ -165,8 +166,6 @@ class ProgressBar
         (epochSize - step + 1) % 60 << "s - loss: " <<
         objective / (double) step <<  "\r";
     output.flush();
-
-    stepTimer.tic();
```

Comment:
I don't think that this is the right solution.  I looked more closely, and I think the reason that the progress bar prediction output is 0 is that `stepTimer.toc()`, which returns a `double`, is being casted immediately to `size_t`.  Often, a single iteration will take far less than 1 second, meaning that `(size_t) stepTimer.toc()` takes a value of `0`, thus leading to a prediction of `0`.  If we avoid casting `stepTimer.toc()` immediately, that should be sufficient (it seems to work in my local tests).

The solution here will actually just cause the timer to accumulate, leading to extremely large and incorrect estimates for the rest of the epoch.

Also, it may be better to use an exponentially weighted average of recent computation times instead of an instantaneous guess, but we can leave that for a future PR or issue. :)

Reply:
Kindly have a look @rcurtin I have incorporated these changes. Thanks.